plugins/node/opentelemetry-instrumentation-knex/src/instrumentation.ts

```diff
@@ -131,13 +131,19 @@ export class KnexInstrumentation extends InstrumentationBase<any> {
     return function wrapQuery(original: () => any) {
       return function wrapped_logging_method(this: any, query: any) {
         const config = this.client.config;
+        const connection = utils.parseConnectionString(
```

Comment:
Maybe we could pass the entire config.connection here and return the resulting values instead of having undefined checks in the attribute assignment further down?